### PR TITLE
filesystem: T3946: root partition auto resize as a service

### PR DESCRIPTION
**Pull request by andriiandrieiev**
## Change Summary

- Added an optional root-partition-suto-resize service for SystemD
- Name of current resizing script and op command were changed to be consistent with the SystemD service.
- Resizing script was changed to work fine with SystemD

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3946

## Component(s) name

filesystem
## Proposed changes

This PR provides root-partition-auto-resize.service which can auto extend root partition and FS when there is enough free space. The service can be enabled via conf mode and is disabled by default.
Op command "force resize-partition" was changed to "force root-partition-auto-resize" to be more understandable and consistent with naming consideration in root-partition-auto-resize.service.
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/system-option.xml.in [hunk 1]
```diff
@@ -117,6 +117,26 @@
                <valueless/>
              </properties>
            </leafNode>
+           <leafNode name="root-partition-auto-resize">
+             <properties>
+               <help>Enable/disable root partition autoextension while system boot</help>
```
Comment (dmbaturin) [inline interface-definitions/system-option.xml.in]: s/while/on/

File: interface-definitions/system-option.xml.in [hunk 2]
```diff
@@ -117,6 +117,26 @@
...
+                 <completionHelp>
+                   <list>enable disable</list>
```
Comment (dmbaturin) [inline interface-definitions/system-option.xml.in]: We have had a policy against this kind of syntax for a long while. If you see a node whose only values are "enable/disable", "true/false" or similar, it's a legacy that should eventually be converted.

The reason is that the value is just a noise. `set system options root-partition-auto-resize` says everything, and "enable" after it adds nothing new. Moreover, a node with "enable/disable" values has two states with the same effect: "exists with default value" and "does not exist", which complicates the script logic.

I can see where you likely got the idea though: `reboot-on-panic` still uses a "true/false" syntax. It's a very old option though and we just never got to updating its syntax (though we should).

This option should be a valueless node like `beep-when-fully-booted`.

File: interface-definitions/system-option.xml.in (hunk 1 above)
Comment (dmbaturin) [inline interface-definitions/system-option.xml.in]: Also, maybe add a hyphen to "auto-extension" for better readability?

**Review by dmbaturin — CHANGES_REQUESTED** [review]
Otherwise, looks good to me.

**Comment by c-po** [comment]
Hi @andriiandrieiev,

first thank you for this very nice solution - never thought about what services could be used for.

We now have two competing solutions to the same project.

- e8c3b66a17473785044ab1a1515ad1485d0870b2 - An op-mode command required to be executed by the user
- This PR adding a CLI option to automatically taking care about resizing

In my opinion - as this is not a daily task - one solution should stick, two are overkill.

To be honest I am no fan of either of them. As VyOS is an appliance it should be treated as one - meaning that if a users extends the disk - on the next reboot it should be automatically resized no matter what. A second reboot could be added just to be sure for consistency reasons.

**Comment by dmbaturin** [comment]
@c-po People generally dislike things happening to their software without their control, and for a good reason.
I think people should have an option to either enable auto-resize and give control to the software, or do it by hand when they think it's safe.

**Comment by c-po** [comment]
Then if I have a choice as user, I would prefer the op-mode command and not have multiple ways.

**Comment by dmbaturin** [comment]
I see no harm in having two options for it. It's the same script in the end, just called either automatically or by hand.

If there were completely different implementations for the manual and the automated process, it definitely would be problematic.

**Review by c-po — APPROVED** [review]

File: interface-definitions/system-option.xml.in (hunk 2 above)
Comment (andriiandrieiev) [reply interface-definitions/system-option.xml.in]: Got it. Fully agree with you. I'll change it.

File: interface-definitions/system-option.xml.in (hunk 1 above)
Comment (andriiandrieiev) [reply interface-definitions/system-option.xml.in]: Sure

**Comment by andriiandrieiev** [comment]
I did all requested changes and pushed an another commit https://github.com/vyos/vyos-1x/commit/bfc2d162e0ca07d7d2a9fe4967690b6399057c3d

**Pull request merged**